deploy/charts/rook-ceph/templates/securityContextConstraints.yaml

```
@@ -0,0 +1,95 @@
+allowPrivilegedContainer: true
+allowHostDirVolumePlugin: true
+allowHostPID: false
+# set to true if running rook with host networking enabled
+allowHostNetwork: false
+# set to true if running rook with the provider as host
+allowHostPorts: false
+priority:
+allowedCapabilities: ["MKNOD"]
+allowHostIPC: true
+readOnlyRootFilesystem: false
+# drop all default privileges
+requiredDropCapabilities: ["All"]
+defaultAddCapabilities: []
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: MustRunAs
+fsGroup:
+  type: MustRunAs
+supplementalGroups:
+  type: RunAsAny
+volumes:
+  - configMap
+  - downwardAPI
+  - emptyDir
+  - hostPath
+  - persistentVolumeClaim
+  - projected
+  - secret
+users:
+  # A user needs to be added for each rook service account.
+  # This assumes running in the default sample "rook-ceph" namespace.
+  # If other namespaces or service accounts are configured, they need to be updated here.
+  - system:serviceaccount:{{ .Values.operatorNamespace | default .Release.Namespace }}:rook-ceph-system # serviceaccount:namespace:operator
+  - system:serviceaccount:{{ .Values.operatorNamespace | default .Release.Namespace }}:default # serviceaccount:namespace:cluster
+  - system:serviceaccount:{{ .Values.operatorNamespace | default .Release.Namespace }}:rook-ceph-mgr # serviceaccount:namespace:cluster
+  - system:serviceaccount:{{ .Values.operatorNamespace | default .Release.Namespace }}:rook-ceph-osd # serviceaccount:namespace:cluster
+  - system:serviceaccount:{{ .Values.operatorNamespace | default .Release.Namespace }}:rook-ceph-rgw # serviceaccount:namespace:cluster
+---
+# scc for the CSI driver
+kind: SecurityContextConstraints
+apiVersion: security.openshift.io/v1
+metadata:
+  name: rook-ceph-csi
+# To allow running privilegedContainers
+allowPrivilegedContainer: true
+# CSI daemonset pod needs hostnetworking
+allowHostNetwork: true
+# This need to be set to true as we use HostPath
+allowHostDirVolumePlugin: true
+priority:
+# SYS_ADMIN is needed for rbd to execute rbd map command
+allowedCapabilities: ["SYS_ADMIN"]
+# Needed as we run liveness container on daemonset pods
+allowHostPorts: true
+# Needed as we are setting this in RBD plugin pod
+allowHostPID: true
+# Required for encryption
+allowHostIPC: true
+# Set to false as we write to RootFilesystem inside csi containers
+readOnlyRootFilesystem: false
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: RunAsAny
+fsGroup:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+# The type of volumes which are mounted to csi pods
+volumes:
+  - configMap
+  - projected
+  - emptyDir
+  - hostPath
+users:
+  # A user needs to be added for each rook service account.
+  # This assumes running in the default sample "rook-ceph" namespace.
+  # If other namespaces or service accounts are configured, they need to be updated here.
+  - system:serviceaccount:{{ .Values.operatorNamespace | default .Release.Namespace }}:rook-csi-rbd-plugin-sa # serviceaccount:namespace:operator
```

```suggestion
  - system:serviceaccount:{{ .Release.Namespace }}:rook-csi-rbd-plugin-sa # serviceaccount:namespace:operator
```

We don't have operatorNamespace in values for operator installation, we have it for ceph cluster creation. these service accounts are created in operator namespace not dependent on ceph cluster

same change is required for all other places as well.

---

Just a question but what if the `operatorNamespace` specified is different? while cluster creation

`{{- $operatorNamespace := .Values.operatorNamespace | default .Release.Namespace -}}`

---

AFAIK that should not be a problem, this is only for operator creation nothing to do with ceph cluster creation.

---

Okay